bayesian/utility/v2/stack_analyses.py

```diff
@@ -0,0 +1,124 @@
+
+import datetime
+import uuid
+import json
+import logging
+from bayesian.dependency_finder import DependencyFinder
+from bayesian.utility.db_gateway import RdbAnalyses
+from bayesian.utility.v2.backbone_server import BackboneServer
+
+logger = logging.getLogger(__file__)
+
+
+class StackAnalyses():
+    """Implements stack analysis API.
+
+    Implements methods to support stack analyses post and get REST APIs calls.
+    """
+
+    def __init__(self, external_request_id=None, ecosystem=None, manifest_file_info=None,
+                 show_transitive=None):
+        """Initialize params to be used for ."""
+        self.external_request_id = external_request_id
+        self.ecosystem = ecosystem
+        self.manifest_file_info = manifest_file_info
+        self.show_transitive = show_transitive
+
+        # Private members used during post request.
+        self._request_date_str = None
+
+    def post_request(self):
+        """Make stack analyses POST request."""
+        # Generate unique request id using UUID, also record timestamp in readable form
+        self.external_request_id = str(uuid.uuid4().hex)
+        self._request_date_str = str(datetime.datetime.now())
+
+        # Make backbone request
+        status, data = self._make_backbone_request()
+        if status != 200:
+            return status, data
+
+        # Finally save results in RDS and upon success return request id.
+        if RdbAnalyses.save_post_request(request_id=self.external_request_id,
+                                         submit_time=self._request_date_str,
+                                         manifest=self.manifest_file_info,
+                                         deps=data) != 0:
+            return 500, 'Error updating db for request {}'.format(self.external_request_id)
+        else:
+            return 200, {
+                'status': 'success',
+                'submitted_at': self._request_date_str,
+                'id': self.external_request_id
+            }
+
+    def _read_deps_and_packages(self):
+        """Read dependencies and packages information from manifest file content."""
+        deps = {}
+        packages = []
+
+        try:
+            # Dependency finder
+            d = DependencyFinder()
+            deps = d.scan_and_find_dependencies(self.ecosystem, [self.manifest_file_info],
+                                                self.show_transitive)
+
+            # Build package details.
+            resolved = deps.get('result', [{}])[0].get('details', [{}])[0].get('_resolved', None)
+            if resolved is not None:
+                for p in resolved:
+                    packages.append({
+                        'name': p.get('package', ''),
+                        'version': p.get('version', ''),
+                        'dependencies': p.get('deps', [])
+                    })
+
+            return 200, {'deps': deps, 'packages': packages}
```

Comment:
Using http error codes for normal operation would add more confusion. Can we use exceptions to deal with error situations?

Reply:
This is fixed in commit 'Review comment to HTTPError from base classes is fixed.'